hearts/hearts.py

```diff
@@ -13,11 +13,79 @@ def __hash__(self):
 
 
 class Game(object):
-    def __init___(self, players, points_to_win=100):
-        self.points_to_win = points_to_win
+    def __init__(self, players, points_to_win=100):
+        self.max_points = points_to_win
         self.players = players
-        self.round_counter = 1
+        self.rounds = []
+        self.round_number = 1
+
         shuffle(self.players)
+        self.create_round()
+
+    def create_round(self):
+        pass_direction = {0: 'keep', 1: 'left', 2: 'right', 3: 'across'}
+        new_round = Round(self.players, pass_direction[self.round_number % 4])
+        self.rounds.append(new_round)
+
+    @property
+    def scores(self):
+        # Returns a dict of the form {round_number: {Player: score}}
+        scores = {1: {player: 0 for player in self.players}}
+        for index, the_round in enumerate(self.rounds, start=1):
+            if the_round.is_over():
+                scores[index] = the_round.final_scores()
+        return scores
+
+    @property
+    def total_scores(self):
+        # Game --> {Player: int}
+        totals = {player: 0 for player in self.players}
+        for round_score in self.scores.values():
+            for player, player_score in round_score.items():
+                totals[player] += player_score
+        return totals
+
+    def is_over(self):
+        return any(score >= self.max_points for score in self.total_scores.values())
+
+    def upkeep(self):
+        if self.rounds[-1].is_over() and not self.is_over():
+            self.round_number += 1
+            self.create_round()
+
+    def rankings(self):
+        '''
+        Returns a dictionary {Player: int} which ranks players based on
+        total scores when the Game is over.
+        Follows "standard competition ranking ("1224" ranking)".
+        '''
+        rankings = {}
+        scores = sorted(self.total_scores.items(), key=lambda x: x[1])
+
+        previous_score = -1   # Store the previous score and rank in the calculation here.
+        previous_rank = -1    # Actual scores and ranks are assumed to be greater than -1.
+
+        for i in range(len(scores)):
+            (player, score) = scores[i]
+            if score > previous_score:
+                rankings[player] = i + 1
+                previous_rank = i + 1
+            else:
+                rankings[player] = previous_rank
+            previous_score = score
+
+        return rankings
+
+    def serialize(self):
+        return {
+            'players': self.players,
+            'max_points': self.max_points,
+            'rounds': [the_round.serialize() for the_round in self.rounds],
+            'round_number': self.round_number,
+            'scores': self.scores,
+            'total_scores': self.total_scores,
+            'is_over': str(self.is_over())
+        }
 
```

Comment:
I made a new issue https://github.com/j2kun/lonely-hearts/issues/58 for this comment and will address it in a separate PR.